pkg/controller/core/clusterqueue_controller.go

```diff
@@ -291,18 +295,47 @@ func (r *ClusterQueueReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *ClusterQueueReconciler) Status(cq *kueue.ClusterQueue) (kueue.ClusterQueueStatus, error) {
+func (r *ClusterQueueReconciler) UpdateCQStatusIfChanged(
+	ctx context.Context,
+	cq *kueue.ClusterQueue,
+	conditionStatus metav1.ConditionStatus,
+	reason, msg string,
+) error {
 	usage, workloads, err := r.cache.Usage(cq)
 	if err != nil {
 		r.log.Error(err, "Failed getting usage from cache")
 		// This is likely because the cluster queue was recently removed,
 		// but we didn't process that event yet.
-		return kueue.ClusterQueueStatus{}, err
+		return err
 	}
+	pendingWorkloads := r.qManager.Pending(cq)
+	oldStatus := cq.Status.DeepCopy()
+	cq.Status.UsedResources = usage
+	cq.Status.AdmittedWorkloads = int32(workloads)
+	cq.Status.PendingWorkloads = int32(pendingWorkloads)
+
+	activeConditionIsChanged := UpdateCQConditionsIfChanged(&cq.Status.Conditions, metav1.Condition{
+		Type:               kueue.ClusterQueueActive,
+		Status:             conditionStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            msg,
+	})
+	if activeConditionIsChanged ||
```

Comment:
wouldn't a single DeepEqual on the entire Status would be enough?

Reply:
Makes sense.

Reply:
this is still pending.

Reply:
Done.